Sources/HTMLString/HTMLString.swift

```diff
@@ -140,35 +144,70 @@ extension String {
     /// | `&` | `&` | Not an entity |
     ///
 
-    public mutating func removeHTMLEntities() {
-        var searchSubstring = self[startIndex ..< endIndex]
+    public func removeHTMLEntities() -> String {
+        var result = ""
+        var currentIndex = startIndex
 
-        while let delimiterIndex = searchSubstring.firstIndex(of: "&") {
+        while let delimiterIndex = self[currentIndex...].firstIndex(of: "&") {
             // Avoid unnecessary operations
-            guard let semicolonIndex = searchSubstring[delimiterIndex ..< endIndex].firstIndex(of: ";") else {
-                break
-            }
+            var semicolonIndex = self.index(after: delimiterIndex)
 
             // Parse the last sequence (ex: Fish & chips &amp; sauce -> "&amp;" instead of "& chips &amp;")
-            let lastDelimiterIndex = searchSubstring[delimiterIndex ..< semicolonIndex].lastIndex(of: "&") ?? delimiterIndex
+            var lastDelimiterIndex = delimiterIndex
+
+            while semicolonIndex != endIndex, self[semicolonIndex] != ";" {
+                if self[semicolonIndex] == "&" {
+                    lastDelimiterIndex = semicolonIndex
+                }
+
+                semicolonIndex = self.index(after: semicolonIndex)
+            }
+
+            // Fast path if semicolon doesn't exists in current range
+            if semicolonIndex == endIndex {
+                result.append(contentsOf: self[currentIndex..<semicolonIndex])
+                return result
+            }
 
             let escapableRange = index(after: lastDelimiterIndex) ..< semicolonIndex
-            let replaceableRange = lastDelimiterIndex ... semicolonIndex
             let escapableContent = self[escapableRange]
 
-            let cursorPosition: Index
-            if let unescapedNumber = escapableContent.unescapeAsNumber() {
-                self.replaceSubrange(replaceableRange, with: unescapedNumber)
-                cursorPosition = self.index(lastDelimiterIndex, offsetBy: unescapedNumber.count)
-            } else if let unescapedCharacter = HTMLStringMappings.shared.unescapingTable[String(escapableContent)] {
-                self.replaceSubrange(replaceableRange, with: unescapedCharacter)
-                cursorPosition = self.index(lastDelimiterIndex, offsetBy: unescapedCharacter.count)
+            result.append(contentsOf: self[currentIndex..<lastDelimiterIndex])
+
+            if lastDelimiterIndex == startIndex {
```

Comment:
Why do we need to check this?

Reply:
Oh, It is useless check! I don't remember why I put it there. I'll delete it.